c++/triqs_Nevanlinna/CMakeLists.txt

```diff
@@ -31,11 +31,13 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DESTINATION include FILES_MATCHING
 
 # ========= Library Dependencies ==========
 
-find_package(Eigen3 REQUIRED)
+find_package(Eigen3 REQUIRED CONFIG)
 target_link_libraries(${PROJECT_NAME}_c PUBLIC Eigen3::Eigen)
+target_compile_definitions(${PROJECT_NAME}_c PUBLIC EIGEN_HAS_STD_RESULT_OF=0)
 
 find_package(Boost REQUIRED)
 target_link_libraries(${PROJECT_NAME}_c PUBLIC Boost::boost)
+target_compile_definitions(${PROJECT_NAME}_c PUBLIC _HAS_AUTO_PTR_ETC=0)
```

Comment:
@hmenke Can you comment on why this is necessary?

Reply:
There is a usage of `std::unary_function` (remove in C++17) in Boost container_hash that is not correctly guarded before Boost 1.73.0, because [`_HAS_AUTO_PTR_ETC` only exists in MSVC](https://devblogs.microsoft.com/cppblog/c17-feature-removals-and-deprecations/) (as far as I can tell).
https://github.com/boostorg/container_hash/commit/90a0e3663875973909f77062dc274081bf32dc0d